rs/consensus/utils/src/lib.rs

```diff
@@ -283,11 +291,14 @@ pub fn find_lowest_ranked_proposals(pool: &PoolReader<'_>, h: Height) -> Vec<Blo
         .fold(
             (None, Vec::new()),
             |(mut best_rank, mut best_proposals), proposal| {
```

Comment:
nit: I guess we don't really need the `best_rank` argument, since we can compute it from `best_proposals` via `best_proposals.first().map(HasRank::rank)`. Maybe getting read of it would make it slightly more readable?

Reply:
Agreed. In fact, using a fold here is unnecessary. Replaced it with simple for loop.